src/components/views/messages/JumpToDatePicker.tsx

```diff
@@ -0,0 +1,175 @@
+    // as someone is typing out a date, we need to disable when we see keydowns.
+    const onDateValueChange = (e: React.ChangeEvent<HTMLSelectElement | HTMLInputElement>): void => {
+        console.log('onDateValueChange')
+        setDateValue(e.target.value);
+
+        // Don't auto navigate if they were manually typing out a date
+        if(navigateOnDatePickerSelection) {
+        onDatePicked(dateValue);
+        }
+    };
+
+    const onDateInputKeyDown = (e: React.KeyboardEvent): void => {
+        // Don't interfere with input default keydown behaviour. For example,
+        // without this, when pressing "Space" while in the input, it would
+        // scroll down the timeline.
+        e.stopPropagation();
+
+        // Ignore the tab key which is probably just navigating focus around
+        // with the keyboard
+        if(e.key === "Tab") {
+            return;
+        }
+
+        // Go and navigate if they submitted
+        if(e.key === "Enter") {
+            onDatePicked(dateValue);
+            return;
+        }
+
+        // When we see someone manually typing out a date, disable the auto
+        // submit on change.
+        setNavigateOnDatePickerSelection(false);
+    };
+
+    const onFormKeyDown = (e: React.KeyboardEvent) => {
+        // Stop the ContextMenu from closing when we first tab from the form to
+        // the datePicker.
+        if(e.target === e.currentTarget && e.key === Key.TAB) {
+            e.stopPropagation();
+        }
+    }
+
+    const onGoKeyDown = (e: React.KeyboardEvent) => {
+        // Stop the ContextMenu from closing when we tab backwards to the
+        // datePicker from the "Go" button
+        //
+        // If they tab forwards off the end of the ContextMenu, we want to close
+        // the ContextMenu which will put the focus back where we were before
+        // opening the ContextMenu.
+        if (e.key === Key.TAB && e.shiftKey) {
+            e.stopPropagation();
+        }
+    }
+
+    const onJumpToDateSubmit = (): void => {
+        onDatePicked(dateValue);
+    }
+
+    return (
+        <form
+            className="mx_JumpToDatePicker_form"
+            onSubmit={onJumpToDateSubmit}
+            onKeyDown={onFormKeyDown}
+            onFocus={onFocus}
+            ref={ref}
+            tabIndex={isActive ? 0 : -1}
+        >
+            <span className="mx_JumpToDatePicker_label">Jump to date</span>
+            <Field
+                element={CustomInput}
+                type="date"
+                onChange={onDateValueChange}
+                onInput={onDateValueInput}
+                onKeyDown={onDateInputKeyDown}
+                value={dateValue}
+                className="mx_JumpToDatePicker_datePicker"
+                label={_t("Pick a date to jump to")}
+                // onFocus={onFocus}
+                // inputRef={ref}
+                tabIndex={isActive ? 0 : -1}
+            />
```

Comment:
Test if can use arrow keys in input field so we don't repeat https://github.com/vector-im/element-web/issues/19881

Reply:
Works ✅